packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,201 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const error = require('../error');
+const isA = require('joi');
+const ScopeSet = require('fxa-shared').oauth.scopes;
+const validators = require('./validators');
+
+module.exports = (log, db, config, customs, push, oauthdb, subhub) => {
+  const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
+    'https://identity.mozilla.com/apps/subscriptions';
+
+  async function handleAuth(auth) {
+    const scope = ScopeSet.fromArray(auth.credentials.scope);
+    if (! scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
+      throw error.invalidToken('Invalid authentication scope in token');
+    }
+    const uid = auth.credentials.user;
+    const account = await db.account(uid);
```

Comment:
Is `email` already on `auth.credentials`?

Reply:
IIRC, it was for a sessionToken but not an oauthToken

Reply:
It looks like `email` should be set for an OAuth request in hapi-fxa-oauth here:

https://github.com/mozilla/hapi-fxa-oauth/blob/master/index.js#L107

The info set there comes from the OAuth server's /verify endpoint here:

https://github.com/mozilla/fxa-auth-server/blob/master/fxa-oauth-server/lib/routes/verify.js#L26

So it might be there and if so, we'd avoid an extra DB lookup here.

Reply:
Hmm, I hadn't been able to put those pieces together before, I think. Went ahead and simplified this

https://github.com/mozilla/fxa/pull/819/files#diff-eedc9e44d94c523680530e9c145c9496R26